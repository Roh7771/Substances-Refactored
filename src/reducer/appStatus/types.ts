import { ModalWindowStatus, ErrorStatus } from '../../types';

export type AppStatusStateType = {
  modalWindowStatus: ModalWindowStatus,
  isSessionChecking: boolean,
  isRequestLoading: boolean,
  errorStatus: ErrorStatus
}

export type AppStatusActionConstsTypes = {
  SET_MODAL_WINDOW_STATUS: 'SET_MODAL_WINDOW_STATUS',
  SET_SESSION_CHECKING_STATUS: 'SET_SESSION_CHECKING_STATUS',
  SET_REQUEST_LOADING_STATUS: 'SET_REQUEST_LOADING_STATUS',
  SET_ERROR_STATUS: 'SET_ERROR_STATUS'
}

export type SetModalWindowStatusActionType = {
  type: AppStatusActionConstsTypes['SET_MODAL_WINDOW_STATUS'],
  payload: ModalWindowStatus
}

export type SetSessionCheckingStatusActionType = {
  type: AppStatusActionConstsTypes['SET_SESSION_CHECKING_STATUS'],
  payload: boolean
}

export type SetRequestLoadingStatusActionType = {
  type: AppStatusActionConstsTypes['SET_REQUEST_LOADING_STATUS'],
  payload: boolean
}

export type SetErrorStatusActionType = {
  type: AppStatusActionConstsTypes['SET_ERROR_STATUS'],
  payload: ErrorStatus
}

export type AppStatusActionTypes = SetModalWindowStatusActionType | SetErrorStatusActionType |
SetSessionCheckingStatusActionType | SetRequestLoadingStatusActionType
