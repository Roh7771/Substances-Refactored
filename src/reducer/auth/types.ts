import { AuthorizationStatus } from '../../const';
import { LoginDataType } from '../../types';

export type AuthStateType = {
  authorizationStatus: AuthorizationStatus;
  csrfToken: string;
};

export type AuthActionConstsTypes = {
  SET_AUTHORIZATION_STATUS: 'SET_AUTHORIZATION_STATUS';
  SET_CSRF_TOKEN: 'SET_CSRF_TOKEN';
  CHECK_SESSION: 'CHECK_SESSION';
  LOGIN: 'LOGIN';
};

export type SetAuthorizationStatusActionType = {
  type: AuthActionConstsTypes['SET_AUTHORIZATION_STATUS'];
  payload: AuthorizationStatus;
};

export type SetCSRFTokenActionType = {
  type: AuthActionConstsTypes['SET_CSRF_TOKEN'];
  payload: string;
};

export type CheckSessionActionType = {
  type: AuthActionConstsTypes['CHECK_SESSION'];
};

export type LoginActionType = {
  type: AuthActionConstsTypes['LOGIN'];
  payload: { loginData: LoginDataType; csrfToken: string };
};

export type AuthActionTypes =
  | SetAuthorizationStatusActionType
  | LoginActionType
  | SetCSRFTokenActionType
  | CheckSessionActionType;
