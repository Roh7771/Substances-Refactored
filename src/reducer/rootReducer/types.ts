import { AuthStateType, AuthActionTypes } from '../auth/types';
import { AppStatusStateType, AppStatusActionTypes } from '../appStatus/types';
import { SubstanceStateType, SubstanceActionTypes } from '../substance/types';

export type CombinedInitialState = {
  authState: AuthStateType,
  appStatusState: AppStatusStateType,
  substancesState: SubstanceStateType,
}

export type CombinedActionTypes = AuthActionTypes | AppStatusActionTypes | SubstanceActionTypes

export type RootReducerType = (state: CombinedInitialState, action: CombinedActionTypes) => CombinedInitialState
