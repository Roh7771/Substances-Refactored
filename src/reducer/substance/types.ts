import {
  LocationCollectionType,
  SubstanceType,
  QueryStringDataType,
} from '../../types';

export type SubstanceStateType = {
  substanceList: SubstanceType[];
  substanceToEdit: SubstanceType | null;
  locationCollection: LocationCollectionType;
  substancesToShow: number;
  queryStringData: QueryStringDataType;
};

export type SubstanceActionConstTypes = {
  SET_SUBSTANCE_LIST: 'SET_SUBSTANCE_LIST';
  SET_SUBSTANCE_TO_EDIT: 'SET_SUBSTANCE_TO_EDIT';
  SET_LOCATION_COLLECTION: 'SET_LOCATION_COLLECTION';
  ADD_SUBSTANCES_TO_SHOW_COUNT: 'ADD_SUBSTANCES_TO_SHOW_COUNT';
  RESET_SUBSTANCES_TO_SHOW_COUNT: 'RESET_SUBSTANCES_TO_SHOW_COUNT';
  SET_QUERY_SEARCH_DATA: 'SET_QUERY_SEARCH_DATA';
  SET_QUERY_LOCATION_DATA: 'SET_QUERY_LOCATION_DATA';
  LOAD_FULL_SUBSTANCE_LIST: 'LOAD_FULL_SUBSTANCE_LIST';
  QUERRY_NEW_DATA: 'QUERRY_NEW_DATA';
  CREATE_SUBSTANCE: 'CREATE_SUBSTANCE';
  UPDATE_SUBSTANCE: 'UPDATE_SUBSTANCE';
  DELETE_SUBSTANCE: 'DELETE_SUBSTANCE';
};

export type SetSubstanceListActionType = {
  type: SubstanceActionConstTypes['SET_SUBSTANCE_LIST'];
  payload: SubstanceType[];
};

export type SetSubstanceToEditActionType = {
  type: SubstanceActionConstTypes['SET_SUBSTANCE_TO_EDIT'];
  payload: SubstanceType | null;
};

export type SetLocationCollectionActionType = {
  type: SubstanceActionConstTypes['SET_LOCATION_COLLECTION'];
  payload: LocationCollectionType;
};

export type AddSubstanceToShowCountActionType = {
  type: SubstanceActionConstTypes['ADD_SUBSTANCES_TO_SHOW_COUNT'];
};

export type ResetSubstanceToShowCountActionType = {
  type: SubstanceActionConstTypes['RESET_SUBSTANCES_TO_SHOW_COUNT'];
};

export type SetQuerySearchDataActionType = {
  type: SubstanceActionConstTypes['SET_QUERY_SEARCH_DATA'];
  payload: {
    type: string;
    value: string;
  };
};

export type SetQueryLocationDataActionType = {
  type: SubstanceActionConstTypes['SET_QUERY_LOCATION_DATA'];
  payload: number[];
};

export type LoadFullSubstanceListActionType = {
  type: SubstanceActionConstTypes['LOAD_FULL_SUBSTANCE_LIST'];
};

export type QuerryNewDataActionType = {
  type: SubstanceActionConstTypes['QUERRY_NEW_DATA'];
  payload: QueryStringDataType;
};

export type CreateSubstanceActionType = {
  type: SubstanceActionConstTypes['CREATE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substance: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type UpdateSubstanceActionType = {
  type: SubstanceActionConstTypes['UPDATE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substance: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type DeleteSubstanceActionType = {
  type: SubstanceActionConstTypes['DELETE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substanceToDelete: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type SubstanceActionTypes =
  | SetSubstanceListActionType
  | SetSubstanceToEditActionType
  | SetLocationCollectionActionType
  | AddSubstanceToShowCountActionType
  | ResetSubstanceToShowCountActionType
  | SetQuerySearchDataActionType
  | SetQueryLocationDataActionType
  | LoadFullSubstanceListActionType
  | QuerryNewDataActionType
  | CreateSubstanceActionType
  | UpdateSubstanceActionType
  | DeleteSubstanceActionType;
