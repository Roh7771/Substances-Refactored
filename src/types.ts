export type SubstanceType = {
  name: string,
  amount?: string,
  casNumber?: string,
  location: number,
  place: string,
  number: number,
  _id?: number,
  company?: string
}

export type LoginDataType = {
  name: string,
  password: string
}

export type QueryStringDataType = {
  search: {
    type: string,
    value: string,
  },
  locations: number[]
}

export type LocationCollectionType = Map<number, Set<string>>

export type AllSubstancesServerResponseType = {
  substances: SubstanceType[];
};

export type CSRFTokenServerResponseType = {
  csrfToken: string;
};

export type OneSubstanceServerResponseType = {
  substance: SubstanceType;
};
