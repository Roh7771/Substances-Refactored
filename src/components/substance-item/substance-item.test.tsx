import React from 'react';
import renderer from 'react-test-renderer';
import { SubstanceType } from '../../types';
import SubstanceItem from './substance-item';

const mockSubstance: SubstanceType = {
  location: 111,
  name: 'some name',
  number: 1,
  place: 'some place',
  _id: 4,
};

it('Substance item should render correctly', () => {
  const tree = renderer.create(<SubstanceItem substance={mockSubstance} />);
  expect(tree).toMatchSnapshot();
});
