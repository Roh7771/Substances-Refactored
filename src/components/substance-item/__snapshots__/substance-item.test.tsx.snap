// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Substance item should render correctly 1`] = `
<tr>
  <th
    className="align-middle"
    scope="row"
  >
    1
  </th>
  <td
    className="align-middle"
  >
    some name
  </td>
  <td
    className="align-middle"
  >
    -
  </td>
  <td
    className="align-middle"
  >
    111
  </td>
  <td
    className="align-middle"
  >
    some place
  </td>
  <td
    className="align-middle"
  >
    -
  </td>
  <td
    className="align-middle"
  >
    -
  </td>
  <td
    className="d-flex flex-column justify-content-center align-items-center"
  >
    <button
      className="btn btn-primary mb-1"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-pen fa-w-16 "
        data-icon="pen"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash fa-w-14 "
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </td>
</tr>
`;
